assets/js/theme/common/product-details.js

```diff
@@ -56,6 +56,27 @@ export default class ProductDetails {
         this.previewModal = modalFactory('#previewModal')[0];
     }
 
+    /**
+     * https://stackoverflow.com/questions/49672992/ajax-request-fails-when-sending-formdata-including-empty-file-input-in-safari
+     * Safari browser with jquery 3.3.1 has an issue uploading empty file parameters. This function removes any empty files from the form params
+     * @param form: FormData object
+     * @returns FormData object
+     */
+    filterEmptyFilesFromForm(form) {
+        try {
+            for (const [key, val] of form.entries()) {
```

Comment:
looks like you should be able to use `const [key, val] of form`

`
An object implementing FormData can directly be used in a for...of structure, instead of entries(): for (var p of myFormData) is equivalent to for (var p of myFormData.entries()).
` (https://developer.mozilla.org/en-US/docs/Web/API/FormData).  maybe rename the param to formData to be explicit?

Reply:
i dont think safaris supports for of on formdata